Library/Homebrew/hardware.rb

```diff
@@ -55,6 +55,8 @@ def self.cores_as_words
     when 1 then "single"
     when 2 then "dual"
     when 4 then "quad"
+    when 6 then "hexa"
```

Comment:
> If there are really no good terms that are universally understood, maybe it's best to actually backtrack a little bit. How about we replace the existing logic with either of these three outputs?

To be fair: I think any of these terms would be understood now so I'm 👍 to 🚢ping as-is.